Add 2D loss map computation using pmap (#267)
CliMA/CalibrateEDMF.jl: experiments/grid_search/README.md

## Performing a grid search and calculating loss maps over pairs of parameters for the TurbulenceConvection.jl EDMF

Authors: Haakon Ludvig Langeland Ervik, Yair Cohen

Last updated : April 2022

> This tool relies on a `config.jl` used in a calbration process. To run this, the `config.jl` file may need to be modified with the following sub-dictionary:

Reviewer: ```suggestion
This tool relies on a `config.jl` used in a calbration process. To run this, the following sub-dictionary needs to be added to the `config.jl` file:
```

```
function get_grid_search_config()
    config = Dict()

>     # Grid search is performed over each pair of parameters, across all specified values

Reviewer: From my point of view, showing a grid search example with 2 parameters can be misleading. User may interpret that if they add another parameter, they will get a cube (in fact, this is what grid search typically refers to). But my understanding is that if I were to add another parameter, you would only compute 2D slices of the cube, centered about the mean of the cube?

It would be enlightening to change the example in this README to 3 parameters, even if you end up running the example only with 2 for computational efficiency.

    config["ensemble_size"] = 1
    # grid search output data stored in `<output_root_dir>/output/YYmmdd_abc`
    config["output_root_dir"] = pwd()
    # Perform grid search and loss map calculation for either the `reference` or `validation` set.
    config["sim_type"] = "reference"  # `reference` or `validation`
    return config
end
```
additionally, add the following line to the `get_config()` function in `config.jl`:
```
# grid search configuration
config["grid_search"] = get_grid_search_config()
```

In the grid search config, 
- `parameters` is a dictionary that specifies the parameters that will be used to generate 2D loss maps, along with the values for each parameter.
- `ensemble_size` is an integer which specifies the number of times a specific simulation configuration is repeated (typically useful for stochastic models)
- `output_root_dir` specifies the parent directory to a directory called `output`, in which you may store all your grid search output. A specific grid search run is stored in a folder identified by today's date + a random three-letter string `YYmmdd_abc`. The full output path is therefore `<output_root_dir>/output/YYmmdd_abc`.

> - `sim_type` is `"reference"` is grid search is run over the `reference` configurations or `"validation"` if it is run over the validation configurations.

Reviewer: ```suggestion
- `sim_type` selects which dataset to use for the map: choose `"reference"` to run a grid search over the `reference` configurations, or `"validation"` to run a grid search over the validation configurations.
```

Running a grid search generates an output folder in which the raw `TurbulenceConvection.jl` data is stored in a systematic way, as well as the `config.jl` file used to generate it.
If you are on the Caltech Central Cluster, you can run grid search by adding it to the schedule:

  >> sbatch run_grid_search.sbatch path/to/config.jl

In case some folders are missing due to an incomplete grid search, you can restart the grid search by

>   >> sbatch restart_grid_search.sbatch path/to/output/YYmmdd_abc

Reviewer: Awesome!
